client/my-sites/plan-features-2023-grid/components/features.tsx

```diff
@@ -92,10 +104,33 @@ const PlanFeatures2023GridFeatures: React.FC< {
 					return null;
 				}
 
-				const isHighlightedFeature = selectedFeature
-					? currentFeature.getSlug() === selectedFeature
-					: currentFeature.getSlug() === FEATURE_CUSTOM_DOMAIN ||
-					  ! currentFeature.availableForCurrentPlan;
+				const isHighlightedFeature = () => {
+					if ( selectedFeature ) {
+						return currentFeature.getSlug() === selectedFeature;
+					}
+
+					if (
+						currentFeature.getSlug() === FEATURE_CUSTOM_DOMAIN ||
+						! currentFeature.availableForCurrentPlan
+					) {
+						return true;
+					}
+
+					const isNewsletter = isNewsletterFlow( flowName ?? null );
+					const isHighlightedNewsletterFeature = [
+						FEATURE_NEWSLETTER_IMPORT_SUBSCRIBERS_FREE,
+						FEATURE_UNLIMITED_SUBSCRIBERS,
+						FEATURE_PAYMENT_TRANSACTION_FEES_4,
+						FEATURE_PAYMENT_TRANSACTION_FEES_8,
+						FEATURE_PAYMENT_TRANSACTION_FEES_10,
+					].includes( currentFeature.getSlug() );
```

Comment:
We are constantly trying to simpliy logic in the grid components (things under `plan-features-2023-grid`). This is introducing flow/intent specific conditioning that I feel should be abstracted out. 

Can you rely on existing `selectedFeature` parameter for this? (maybe turn that into an array of slugs and pass these Newsletter-specific through `plans-features-main` (or the flow step if you only need to highlight from Signup).

Reply:
Hey - @chriskmnds - Thanks for the input!

Yes, there's a separation of concerns issue with adding the feature-specific and flow-specific logic here as I have. So happy to look at changing that. I thought about it but was trying to avoid the immediate complexity of adding the needed abstraction. 

If you'd like me to use the selectedFeature prop, I can. But it doesn't seem intended for this use case? And I do think there's a repetitive need to be able to designate certain features on a plans list as highlighted. 

There is already a basis for doing this in the file isHighlightedFeature. But isHighlightedFeature was, itself, implemented in a very specific way to handle selectedFeature and one specific feature, FEATURE_CUSTOM_DOMAIN. 

We could convert that to a method with more generic logic using something like [getHighlightedFeatures](https://github.com/Automattic/wp-calypso/blob/f1631c3c1951bc2a13bf9e9671b38230b4bcca6c/client/my-sites/plan-features-comparison/util.ts#L76C14-L76C36). A starting point for now might be something like:

```
const isHighlightedFeature = () => 
  // Keep the existing logic
  currentFeature.getSlug() === selectedFeature ||
  currentFeature.getSlug() === FEATURE_CUSTOM_DOMAIN ||
  ! currentFeature.availableForCurrentPlan || 
  // And just add this line
  getHighlightedFeatures( flowName, plan ).includes( currentFeature.getSlug() )
}
```
Of course, this still smuggles in some flow-specific logic for determining which features are highlighted. 

On other approach I considered is adding an attribute to each feature itself. The main issue with this is that which features get highlighted probably does depend on the context in terms of flow or intent. But in any case, if you look at [this file and line](https://github.com/Automattic/wp-calypso/blob/f1631c3c1951bc2a13bf9e9671b38230b4bcca6c/client/lib/plans/features-list.tsx#L1636), update that to something like:

```
[ FEATURE_UNLIMITED_SUBSCRIBERS ]: {
    getSlug: () => FEATURE_UNLIMITED_SUBSCRIBERS,
    getTitle: () => i18n.translate( 'Import unlimited subscribers' ),
    isHighlighted: true
 },
 ```
 
Then the isHighlightedFeature variable could be updated to: 

```
const isHighlightedFeature = () => 
  // Just add this one extra line of logic
  // Using optional chaining means we don't need to worry about adding this prop to all features
  currentFeature?.isHighlighted || 
  currentFeature.getSlug() === selectedFeature ||
  // Could get rid of this line by making .isHighlighted === true for FEATURE_CUSTOM_DOMAIN
  currentFeature.getSlug() === FEATURE_CUSTOM_DOMAIN ||
  ! currentFeature.availableForCurrentPlan || 
}
```

Let me know if you have thoughts!

Reply:
@chriskmnds - I've prepped an alternative PR for this here: https://github.com/Automattic/wp-calypso/pull/81418

Some notes. First, on using the selectedFeature prop...
   - There's a comment [here](https://github.com/Automattic/wp-calypso/blob/1e3a62f75544236a3c07ab30dd5fe95476e039f9/client/my-sites/plan-features-2023-grid/index.tsx#L87) that selectedFeature is the 'value of the ?feature= query param, so we can highlight a given feature and hide plans without it.' That seems like a fairly specific purpose, a bit different than what we're trying to here, which was why I mentioned that above.
    - We'd need to update that one param to handle both a string derived from a query param and an array passed in as prop. There may be some other differences too - like we don't want to hide plans without it (if indeed that's being done). 
    - A related concern was that the selectedFeature appears in quite a few files. We'd need to refactor all the associated components and methods that use it to handle array vs string. 
   
On the use-plan-features-for-grid-plans hook and TransformedFeatureObject... 
   - This did seems promising, and this is where I tried my new implementation. It's worth noting that this [lives within the plans-features-grid-2023 folder](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-plan-features-for-grid-plans.ts), and you mentioned wanting to address this outside that folder. Sounds like you'd be ok adjusting logic for this particular file?